File: GildedRose/src/com/gildedrose/tests/RegularItemTest.java
package com.gildedrose.tests;

import com.gildedrose.items.*;
import com.gildedrose.GildedRose;

import org.junit.Test;
import static org.junit.Assert.*;


public class RegularItemTest {

    @Test
    public void updateQuality_SellinGreterThanZero_QualityDecreaseByOne() {
    	//Arrange
        Item[] items = new Item[] { new Item("+5 Dexterity Vest", 10, 20) };
        GildedRose app = new GildedRose(items);
        
        //Act
        app.updateQuality();
        
        //Assert
        assertEquals("+5 Dexterity Vest", items[0].name);
        assertEquals(9, items[0].sellIn);
        assertEquals(19, items[0].quality);
    }

    @Test
    public void updateQuality_SellinEqualsZero_QualityDecreaseByTwo() {
    	//Arrange
        Item[] items = new Item[] { new Item("+5 Dexterity Vest", 0, 20) };
        GildedRose app = new GildedRose(items);
        
        //Act
        app.updateQuality();
        
        //Assert
        assertEquals("+5 Dexterity Vest", items[0].name);
        assertEquals(-1, items[0].sellIn);
        assertEquals(18, items[0].quality);
    }

    @Test
    public void updateQuality_SellinLessThanZero_QualityDecreaseByTwo() {
    	//Arrange
        Item[] items = new Item[] { new Item("+5 Dexterity Vest", -1, 20) };
        GildedRose app = new GildedRose(items);
        
        //Act
        app.updateQuality();
        
        //Assert
        assertEquals("+5 Dexterity Vest", items[0].name);
        assertEquals(-2, items[0].sellIn);
        assertEquals(18, items[0].quality);
    }

    @Test
    public void updateQuality_QualityEqualsZero_QualityUnchanged() {
    	//Arrange
        Item[] items = new Item[] { new Item("+5 Dexterity Vest", 10, 0) };
        GildedRose app = new GildedRose(items);
        
        //Act
        app.updateQuality();
        
        //Assert
        assertEquals("+5 Dexterity Vest", items[0].name);
        assertEquals(9, items[0].sellIn);
        assertEquals(0, items[0].quality);
    }

    @Test
    public void updateQuality_QualityEqualsFifty_QualityDecreaseByOne() {
    	//Arrange
        Item[] items = new Item[] { new Item("+5 Dexterity Vest", 10, 50) };
        GildedRose app = new GildedRose(items);
        
        //Act
        app.updateQuality();
        
        //Assert
        assertEquals("+5 Dexterity Vest", items[0].name);
        assertEquals(9, items[0].sellIn);
        assertEquals(49, items[0].quality);
    }

}

File: GildedRose.Net/Tests/RegularItemTest.cs
using Xunit;

using GildedRose.Items;

namespace GildedRose.Tests
{
    public class RegularItemTest
    {
        
        [Fact]
        public void updateQuality_SellinGreterThanZero_QualityDecreaseByOne()
        {
            //Arrange
            Item[] items = new Item[] { new Item{Name = "+5 Dexterity Vest", SellIn = 10, Quality = 20} };
            GildedRose app = new GildedRose(items);
            
            //Act
            app.UpdateQuality();
            
            //Assert
            Assert.Equal("+5 Dexterity Vest", items[0].Name);
            Assert.Equal(9, items[0].SellIn);
            Assert.Equal(19, items[0].Quality);
        }
    
        [Fact] 
        public void updateQuality_SellinEqualsZero_QualityDecreaseByTwo() 
        {
            //Arrange
            Item[] items = new Item[] { new Item{Name = "+5 Dexterity Vest", SellIn = 0, Quality = 20} };
            GildedRose app = new GildedRose(items);
            
            //Act
            app.UpdateQuality();
            
            //Assert
            Assert.Equal("+5 Dexterity Vest", items[0].Name);
            Assert.Equal(-1, items[0].SellIn);
            Assert.Equal(18, items[0].Quality);
        }
    
        [Fact]
        public void updateQuality_SellinLessThanZero_QualityDecreaseByTwo() 
        {
            //Arrange
            Item[] items = new Item[] { new Item{Name = "+5 Dexterity Vest", SellIn = -1, Quality = 20} };
            GildedRose app = new GildedRose(items);
            
            //Act
            app.UpdateQuality();
            
            //Assert
            Assert.Equal("+5 Dexterity Vest", items[0].Name);
            Assert.Equal(-2, items[0].SellIn);
            Assert.Equal(18, items[0].Quality);
        }
    
        [Fact]
        public void updateQuality_QualityEqualsZero_QualityUnchanged() 
        {
            //Arrange
            Item[] items = new Item[] { new Item{Name = "+5 Dexterity Vest", SellIn = 10, Quality = 0} };
            GildedRose app = new GildedRose(items);
            
            //Act
            app.UpdateQuality();
            
            //Assert
            Assert.Equal("+5 Dexterity Vest", items[0].Name);
            Assert.Equal(9, items[0].SellIn);
            Assert.Equal(0, items[0].Quality);
        }
    
        [Fact]
        public void updateQuality_QualityEqualsFifty_QualityDecreaseByOne() 
        {
            //Arrange
            Item[] items = new Item[] { new Item{Name = "+5 Dexterity Vest", SellIn = 10, Quality = 50} };
            GildedRose app = new GildedRose(items);
            
            //Act
            app.UpdateQuality();
            
            //Assert
            Assert.Equal("+5 Dexterity Vest", items[0].Name);
            Assert.Equal(9, items[0].SellIn);
            Assert.Equal(49, items[0].Quality);
        }

    }
}
